src/connection.cpp

```diff
@@ -495,6 +489,19 @@ namespace oxen::quic
         return 0;
     }
 
+    void Connection::check_established()
+    {
+        if (establish_hook_called)
```

Comment:
both call sites are also doing this check, so it's slightly redundant (but probably good to keep anyway?)